util/src/main/scala/org/ensime/vfs/EnsimeVFS.scala

```diff
@@ -46,6 +46,20 @@ object `package` {
   private val zipFileField = classOf[ZipFileSystem].getDeclaredField("zipFile")
   zipFileField.setAccessible(true)
 
+  object URLParamEncoder {
+    private val unsafe = " %"
+    def encode(input: String): String = {
+      val resultStr = new StringBuilder()
+      input.toCharArray.foreach { ch =>
+        if (unsafe contains ch)
```

Comment:
I'd just use an `if` statement to be honest. The object creation that happens when `String` is treated as a `Traversable` is pretty crazy.